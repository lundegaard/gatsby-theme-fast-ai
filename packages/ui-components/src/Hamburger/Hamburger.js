import React from 'react';
import PropTypes from 'prop-types';

import Flex from '../Flex';
import Box from '../Box';

const Hamburger = ({ isOpen, ...rest }) => (
	<Flex
		sx={{
			transition: 'background-color .25s ease-in',
			backgroundColor: isOpen ? 'primary' : 'transparent',
			height: 64,
			width: 64,
			alignItems: 'center',
			justifyContent: 'center',
			cursor: 'pointer',
		}}
		{...rest}
	>
		<Box
			sx={{
				position: 'relative',
				width: '24px',
				height: '18px',
				transform: 'rotate(0deg)',
				transition: '.3s ease-in-out',

				span: {
					display: 'block',
					position: 'absolute',
					height: '2px',
					width: '100%',
					backgroundColor: 'primary',
					opacity: 1,
					transform: 'translateX(0)',
					transition: 'transform .25s ease-in, opacity .25s ease-in',
				},

				'span:nth-child(1)': {
					top: 0,
					...(isOpen
						? {
								backgroundColor: 'white',
								transform: 'translateX(200%)',
								opacity: 0,
						  }
						: {}),
				},

				'span:nth-child(2)': {
					top: 'calc(( 100% / 2 ) - 1px )',
					...(isOpen
						? {
								transform: 'rotate(-45deg)',
								backgroundColor: 'white',
						  }
						: {}),

					'&:after': {
						content: '""',
						display: 'block',
						position: 'absolute',
						height: '2px',
						width: '100%',
						backgroundColor: 'primary',
						opacity: 0,
						left: 0,
						transform: 'rotate(0) translateX(0)',
						transition: 'opacity .25s ease-in, transform .25s ease-in',
						...(isOpen
							? {
									transform: 'rotate(90deg)',
									opacity: 1,
									backgroundColor: 'white',
							  }
							: {}),
					},
				},

				'span:nth-child(3)': {
					bottom: 0,
					...(isOpen
						? {
								backgroundColor: 'white',
								transform: 'translateX(200%)',
								opacity: 0,
						  }
						: {}),
				},
			}}
		>
			<span />
			<span />
			<span />
		</Box>
	</Flex>
);

Hamburger.propTypes = { isOpen: PropTypes.bool };

export default Hamburger;
