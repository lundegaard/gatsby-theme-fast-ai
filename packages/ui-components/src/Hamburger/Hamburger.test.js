import React from 'react';
import { shallow } from 'enzyme';

import Hamburger from './Hamburger';

describe('<Hamburger />', () => {
	const requiredProps = {};

	it('matches snapshot when shallowly rendered', () => {
		const wrapper = shallow(<Hamburger {...requiredProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
