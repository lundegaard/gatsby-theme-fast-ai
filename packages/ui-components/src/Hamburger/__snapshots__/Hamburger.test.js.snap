// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hamburger /> matches snapshot when shallowly rendered 1`] = `
<Flex
  sx={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "cursor": "pointer",
      "height": 64,
      "justifyContent": "center",
      "transition": "background-color .25s ease-in",
      "width": 64,
    }
  }
>
  <Box
    sx={
      Object {
        "height": "18px",
        "position": "relative",
        "span": Object {
          "backgroundColor": "primary",
          "display": "block",
          "height": "2px",
          "opacity": 1,
          "position": "absolute",
          "transform": "translateX(0)",
          "transition": "transform .25s ease-in, opacity .25s ease-in",
          "width": "100%",
        },
        "span:nth-child(1)": Object {
          "top": 0,
        },
        "span:nth-child(2)": Object {
          "&:after": Object {
            "backgroundColor": "primary",
            "content": "\\"\\"",
            "display": "block",
            "height": "2px",
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "transform": "rotate(0) translateX(0)",
            "transition": "opacity .25s ease-in, transform .25s ease-in",
            "width": "100%",
          },
          "top": "calc(( 100% / 2 ) - 1px )",
        },
        "span:nth-child(3)": Object {
          "bottom": 0,
        },
        "transform": "rotate(0deg)",
        "transition": ".3s ease-in-out",
        "width": "24px",
      }
    }
  >
    <span />
    <span />
    <span />
  </Box>
</Flex>
`;
